### ISSUE TITLE
Always generates empty dependencies since upgrade

### ISSUE THREAD

**Issue by pixelplanetdev** [lostpebble/generate-package-json-webpack-plugin#27]
I recently upgraded my working configuration from generate-package-json-webpack-plugin 1.1.2 and  webpack 4.x to:
``` 
    "babel-loader": "^8.2.2",
    "generate-package-json-webpack-plugin": "^2.1.1",
    "webpack": "^5.9.0",
    "webpack-node-externals": "^2.5.2",
```
and now the generated package.json has always empty dependencies.
[Here the webpack config](https://github.com/pixelplanetdev/pixelplanet/blob/master/tools/webpack.config.web.js)
I first thought it's just some configuration error of mine, but i changed all possible options and always got no dependencies. I tried to drop webpack-node-externals and replace it with an own script, still nothing. The verbose output never finds even a single external module.

verbose log while building:
```
Bundle with webpack....
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].use[0]!/home/andreas/Projekte/pixelplanet/src/styles/globe.css
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/css-loader/dist/runtime/api.js
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ruleSet[1].rules[0]!/home/andreas/Projekte/pixelplanet/src/web.js|66a25a6b1955160fe64cd1d615fa70c1
GPJWP: Found module: webpack/runtime/compat get default export
GPJWP: Found module: webpack/runtime/define property getters
GPJWP: Found module: webpack/runtime/make namespace object
GPJWP: Found module: webpack/runtime/hasOwnProperty shorthand
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ruleSet[1].rules[0]!/home/andreas/Projekte/pixelplanet/src/backup.js|9229503110e29a788fa3f5640c727bd6
GPJWP: Found module: webpack/runtime/make namespace object
GPJWP: Found module: webpack/runtime/compat get default export
GPJWP: Found module: webpack/runtime/define property getters
GPJWP: Found module: webpack/runtime/hasOwnProperty shorthand
GPJWP: Modules to be used in generated package.json {}
```

**Comment by lostpebble** [comment]
Hi @pixelplanetdev ,

I'll look into this soon.

For now, have you tried using the option `useInstalledVersions: true` ? This may fix your problems and is the recommended way to use the plugin now- instead of relying on the source package files. Though that should still work, so need to see why that's not working in this case.

**Comment by lostpebble** [comment]
What's interesting is that you don't seem to be getting a debug log along the lines of:

```
GPJWP: Received ${allSourcePackageFilenames.length} extra package.json file from which to source versions (later takes preference):
```

Which should be showing, since you are using this option and passing in `[path.resolve(__dirname, '../package.json')]`

**Comment by lostpebble** [comment]
Although, if you are never getting the log:

```
GPJWP: Found external module: "..."
```

Then that points to a probable bigger issue. Perhaps the latest version of webpack isn't compatible, although we did make changes to make it work with webpack 5- I wonder if something changed on their side.

**Comment by pixelplanetdev** [comment]
hi, thanks for the fast reply.
Yes, i did try with `useInstalledVersion: true`. Same result.
At the verbose log that i posted i had that one dependency removed to make sure that i have a minimum configuration.

Here again with that one dependency set:
```
GeneratePackageJsonPlugin: Debugging mode activated!
GPJWP: Received 1 extra package.json file from which to source versions (later takes preference):
1 : /home/andreas/Projekte/pixelplanet/package.json
GPJWP: Final map of dependency versions to be used if encountered in bundle:
 { bcrypt: '^5.0.0',
  bluebird: '^3.5.0',
  'body-parser': '^1.17.2',
  bufferutil: '^4.0.1',
  compression: '^1.7.3',
  'connect-redis': '^5.0.0',
  'cookie-parser': '^1.4.5',
  'core-js': '^3.8.0',
  cors: '^2.8.4',
  etag: '^1.8.1',
  express: '^4.15.3',
  'express-limiter': '^1.6.0',
  'express-session': '^1.17.1',
  global: '^4.3.2',
  'http-proxy-agent': '^4.0.1',
  'image-q': '^2.1.2',
  'ip-address': '^7.1.0',
  'isomorphic-fetch': '^3.0.0',
  'js-file-download': '^0.4.12',
  keycode: '^2.1.9',
  localforage: '^1.9.0',
  morgan: '^1.10.0',
  multer: '^1.4.1',
  mysql2: '^2.2.5',
  nodemailer: '^6.4.16',
  passport: '^0.4.0',
  'passport-discord': '^0.1.4',
  'passport-facebook': '^3.0.0',
  'passport-google-oauth': '^2.0.0',
  'passport-json': '^1.2.0',
  'passport-reddit': '^0.2.4',
  'passport-vkontakte': '^0.5.0',
  react: '^17.0.1',
  'react-dom': '^17.0.1',
  'react-icons': '^4.1.0',
  'react-modal': '^3.12.1',
  'react-redux': '^7.2.1',
  'react-responsive': '^8.2.0',
  'react-stay-scrolled': '^7.1.0',
  'react-toggle-button': '^2.1.0',
  redis: '^3.0.2',
  redlock: '^4.0.0',
  redux: '^4.0.4',
  'redux-logger': '^3.0.6',
  'redux-persist': '^6.0.0',
  'redux-thunk': '^2.2.0',
  sequelize: '^6.3.5',
  sharp: '^0.26.3',
  startaudiocontext: '^1.2.1',
  sweetalert2: '^10.10.4',
  three: '^0.123.0',
  'three-trackballcontrols': '^0.9.0',
  'url-search-params-polyfill': '^8.1.0',
  winston: '^3.3.3',
  'winston-daily-rotate-file': '^4.5.0',
  ws: '^7.4.0' }
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].use[0]!/home/andreas/Projekte/pixelplanet/src/styles/globe.css
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/css-loader/dist/runtime/api.js
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ruleSet[1].rules[0]!/home/andreas/Projekte/pixelplanet/src/web.js|66a25a6b1955160fe64cd1d615fa70c1
GPJWP: Found module: webpack/runtime/compat get default export
GPJWP: Found module: webpack/runtime/define property getters
GPJWP: Found module: webpack/runtime/make namespace object
GPJWP: Found module: webpack/runtime/hasOwnProperty shorthand
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ruleSet[1].rules[0]!/home/andreas/Projekte/pixelplanet/src/backup.js|9229503110e29a788fa3f5640c727bd6
GPJWP: Found module: webpack/runtime/make namespace object
GPJWP: Found module: webpack/runtime/compat get default export
GPJWP: Found module: webpack/runtime/define property getters
GPJWP: Found module: webpack/runtime/hasOwnProperty shorthand
GPJWP: Adding deliberate module in "dependencies" with version set deliberately: mysql2 -> ^2.1.0
GPJWP: Adding deliberate module in "dependencies" with version found in sources: mysql2 -> ^2.2.5
GPJWP: Modules to be used in generated package.json { mysql2: '^2.2.5' }
```

**Comment by pixelplanetdev** [comment]
and here with `useInstalledVersion: true` and no `sourcePackageFilenames`

```
GeneratePackageJsonPlugin: Debugging mode activated!
GPJWP: Final map of dependency versions to be used if encountered in bundle:
 {}
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].use[0]!/home/andreas/Projekte/pixelplanet/src/styles/globe.css
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/css-loader/dist/runtime/api.js
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ruleSet[1].rules[0]!/home/andreas/Projekte/pixelplanet/src/web.js|66a25a6b1955160fe64cd1d615fa70c1
GPJWP: Found module: webpack/runtime/compat get default export
GPJWP: Found module: webpack/runtime/define property getters
GPJWP: Found module: webpack/runtime/make namespace object
GPJWP: Found module: webpack/runtime/hasOwnProperty shorthand
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ruleSet[1].rules[0]!/home/andreas/Projekte/pixelplanet/src/backup.js|9229503110e29a788fa3f5640c727bd6
GPJWP: Found module: webpack/runtime/make namespace object
GPJWP: Found module: webpack/runtime/compat get default export
GPJWP: Found module: webpack/runtime/define property getters
GPJWP: Found module: webpack/runtime/hasOwnProperty shorthand
GPJWP: Adding deliberate module in "dependencies" with version set deliberately: mysql2 -> ^2.1.0
GPJWP: Modules to be used in generated package.json { mysql2: '^2.1.0' }
```

**Comment by lostpebble** [comment]
Okay, well then it seems likely that this is an issue with a more recent Webpack version 5... Since it seems you are getting no indication that the plugin is picking up any imports during the build process.

For a quick test can you attempt a build after downgrading to Webpack 4, and see if you get the same outcome? Would help a lot to determine where the potential issue is.

**Comment by pixelplanetdev** [comment]
Here with webpack 4.44.2, where it works.
```
> PixelPlanet@1.0.0 webpack /home/andreas/Projekte/pixelplanet
> webpack --config webpack.config.babel.js

GeneratePackageJsonPlugin: Debugging mode activated!
GPJWP: Final map of dependency versions to be used if encountered in bundle:
 {}
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ref--4!/home/andreas/Projekte/pixelplanet/src/core/constants.js
GPJWP: Found external module: external "fs"
GPJWP: Native node.js module detected: external "fs"
GPJWP: Found external module: external "./canvases.json"
GPJWP: Ignoring module without a found package.json: . ("package.json" couldn't resolve)
GPJWP: Found external module: external "sharp"
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ref--4!/home/andreas/Projekte/pixelplanet/src/core/Palette.js
GPJWP: Found external module: external "bluebird"
GPJWP: Found external module: external "redis"
GPJWP: Found external module: external "path"
GPJWP: Native node.js module detected: external "path"
GPJWP: Found external module: external "process"
GPJWP: Native node.js module detected: external "process"
GPJWP: Found external module: external "child_process"
GPJWP: Native node.js module detected: external "child_process"
GPJWP: Found module: multi ./src/backup.js
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ref--4!/home/andreas/Projekte/pixelplanet/src/backup.js 59ea488ddde45b8e197bbfc1c33596f1
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ref--4!/home/andreas/Projekte/pixelplanet/src/core/constants.js
GPJWP: Found external module: external "sequelize"
GPJWP: Found external module: external "fs"
GPJWP: Native node.js module detected: external "fs"
GPJWP: Found external module: external "./canvases.json"
GPJWP: Ignoring module without a found package.json: . ("package.json" couldn't resolve)
GPJWP: Found external module: external "ws"
GPJWP: Found external module: external "express"
GPJWP: Found external module: external "sharp"
GPJWP: Found external module: external "winston"
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ref--4!/home/andreas/Projekte/pixelplanet/src/core/Palette.js
GPJWP: Found external module: external "bluebird"
GPJWP: Found external module: external "./assets.json"
GPJWP: Ignoring module without a found package.json: . ("package.json" couldn't resolve)
GPJWP: Found external module: external "redis"
GPJWP: Found external module: external "passport"
GPJWP: Found external module: external "path"
GPJWP: Native node.js module detected: external "path"
GPJWP: Found external module: external "react-dom/server"
GPJWP: Found external module: external "body-parser"
GPJWP: Found external module: external "isomorphic-fetch"
GPJWP: Found external module: external "react"
GPJWP: Found external module: external "etag"
GPJWP: Found external module: external "express-limiter"
GPJWP: Found external module: external "winston-daily-rotate-file"
GPJWP: Found external module: external "bcrypt"
GPJWP: Found external module: external "./proxies.json"
GPJWP: Ignoring module without a found package.json: . ("package.json" couldn't resolve)
GPJWP: Found external module: external "compression"
GPJWP: Found external module: external "ip-address"
GPJWP: Found external module: external "express-session"
GPJWP: Found external module: external "nodemailer"
GPJWP: Found external module: external "./styleassets.json"
GPJWP: Ignoring module without a found package.json: . ("package.json" couldn't resolve)
GPJWP: Found external module: external "url"
GPJWP: Native node.js module detected: external "url"
GPJWP: Found external module: external "http"
GPJWP: Native node.js module detected: external "http"
GPJWP: Found external module: external "redlock"
GPJWP: Found external module: external "http-proxy-agent"
GPJWP: Found external module: external "connect-redis"
GPJWP: Found external module: external "passport-json"
GPJWP: Found external module: external "passport-discord"
GPJWP: Found external module: external "passport-reddit"
GPJWP: Found external module: external "passport-facebook"
GPJWP: Found external module: external "passport-vkontakte"
GPJWP: Found external module: external "passport-google-oauth"
GPJWP: Found external module: external "multer"
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/css-loader/dist/cjs.js??ref--5-0!/home/andreas/Projekte/pixelplanet/src/styles/globe.css
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/css-loader/dist/runtime/api.js
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/src/data/countrycode-coords-array.json
GPJWP: Found module: multi ./src/web.js
GPJWP: Found module: /home/andreas/Projekte/pixelplanet/node_modules/babel-loader/lib/index.js??ref--4!/home/andreas/Projekte/pixelplanet/src/web.js 412375b4a2675de07ee8bb9c25eb4d0c
GPJWP: Adding deliberate module in "dependencies" with version set deliberately: mysql2 -> ^2.1.0
GPJWP: Modules to be used in generated package.json { '.': undefined,
  sharp: '0.26.3',
  bluebird: '3.7.2',
  redis: '3.0.2',
  sequelize: '6.3.5',
  ws: '7.4.0',
  express: '4.17.1',
  winston: '3.3.3',
  passport: '0.4.1',
  'react-dom': '17.0.1',
  'body-parser': '1.19.0',
  'isomorphic-fetch': '3.0.0',
  react: '17.0.1',
  etag: '1.8.1',
  'express-limiter': '1.6.1',
  'winston-daily-rotate-file': '4.5.0',
  bcrypt: '5.0.0',
  compression: '1.7.4',
  'ip-address': '7.1.0',
  'express-session': '1.17.1',
  nodemailer: '6.4.16',
  redlock: '4.2.0',
  'http-proxy-agent': '4.0.1',
  'connect-redis': '5.0.0',
  'passport-json': '1.2.0',
  'passport-discord': '0.1.4',
  'passport-reddit': '0.2.4',
  'passport-facebook': '3.0.0',
  'passport-vkontakte': '0.5.0',
  'passport-google-oauth': '2.0.0',
  multer: '1.4.2',
  mysql2: '^2.1.0' }
[webpack-cli] Compilation finished```
i also tried with webpack `5.0.0` and that already doesn't work.

**Comment by pixelplanetdev** [comment]
wait, i didn't include the whole logs with the stats after compilation:
with 4.44.2:
https://pastebin.com/F3RA9K0f
with 5.9.0 (or any 5.x build):
https://pastebin.com/aGgwiW1e
It's talking about orphan modules here.

**Comment by lostpebble** [comment]
Thanks for checking that out @pixelplanetdev !

Okay, so I guess the library at this moment doesn't support Webpack 5 yet. Will need to look into providing proper support for it in the coming days, and what exactly that entails.

**Comment by lostpebble** [comment]
Published 2.1.2 with the fixes
